### PR TITLE
layers: New wsi_validation for CoreChecks

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Creates a new `wsi_validation.cpp` file to move WSI related `CoreChecks` into

### CODE REVIEW

File: layers/core_validation.h [hunk 1]
```diff
@@ -222,6 +223,107 @@ class CORE_CMD_BUFFER_STATE : public CMD_BUFFER_STATE {
                           VkPipelineStageFlags2KHR src_stage_mask) override;
 };
 
+struct TimelineMaxDiffCheck {
```
Comment (ncesario-lunarg) [inline layers/core_validation.h:226]: Seems like this (and _maybe_ SemaphoreSubmitState below) belongs in a utility header (e.g., `vk_layer_utils.h`/`vk_layer_data.h`)?

**Review by ncesario-lunarg — APPROVED** [review]
One question, but if this is again just moving code without other refactoring, LGTM!

File: layers/core_validation.h (hunk 1 above)
Comment (jeremyg-lunarg) [reply layers/core_validation.h:226]: Yes to the possiblity of another header. No to either of those.  SemaphoreSubmitState is heavily tied to core validation and it wouldn't be a good fit for either of those files.

Comment (spencer-lunarg) [reply layers/core_validation.h:226]: I thought about making a new header, but figured it might be better for the short term to keep here near were it was

**Comment by spencer-lunarg** [comment]
> just moving code without other refactoring, LGTM!

The only "code change" was changing the 3 functions from using a template to using the new `LogObjectList` system

```patch
- template <typename T1>
- bool ValidateDeviceMaskToPhysicalDeviceCount(uint32_t deviceMask, const T1 object, const char* VUID) const;
+ bool ValidateDeviceMaskToPhysicalDeviceCount(uint32_t deviceMask, const LogObjectList& objlist, const char* VUID) const;
```

this was done because I was getting 

> undefined symbol: _ZNK10CoreChecks39ValidateDeviceMaskToPhysicalDeviceCountIP16VkSwapchainKHR_TEEbjT_PKc

and also it was on my radar to use `LogObjectList` everywhere instead anyway

**Pull request merged**